src/cartridges/bm_adyen/cartridge/static/default/adyenSettings.js

```diff
@@ -3,84 +3,174 @@ document.addEventListener('DOMContentLoaded', () => {
+      changedSettings[settingIndex] = {
+        key,
+        value
+      };
     } else {
-      changedSettings.push({ key, value });
+      changedSettings.push({
+        key,
+        value
+      });
     }
   }
 
   function enableformButtons() {
-    formButtons.forEach((button) => {
+    for (const button of formButtons) {
       button.classList.remove('disabled');
       button.classList.add('enabled');
       form.removeEventListener('input', enableformButtons);
-    });
+    }
+
+    ;
   }
 
   function disableFormButtons() {
-    formButtons.forEach((button) => {
+    for (const button of formButtons) {
       button.classList.remove('enabled');
       button.classList.add('disabled');
       form.removeEventListener('input', enableformButtons);
-    });
+    }
+  } // add event for save button availability on form change.
+
+
+  function clickAndHide() {
+    document.getElementById('settingsFormSubmitButton').click();
+    document.getElementById('saveChangesAlert').hide();
+    document.getElementById('notSavedChangesAlert').hide();
+    document.getElementById('testConnectionButton').disabled = true;
+  }
+
+  function showAlerts() {
+    document.getElementById('saveChangesAlert').show();
+    document.getElementById('notSavedChangesAlert').show();
   }
 
-  // add event for save button availability on form change.
-  form.addEventListener('input', enableformButtons);
+  function renderSafari(){
+    if (/^((?!chrome|android).)*safari/i.test(navigator.userAgent)){
+      formBody.style.setProperty('padding-top', '3rem');
+    }
+  }
+
+  testConnectionButton.addEventListener('click', clickAndHide); // add event listener to hide the alerts in case of test connection
+
+  form.addEventListener('input', enableformButtons); // add event listener to maintain form updates
+
+  submitButton.addEventListener('click', showAlerts); // add event listener to show alerts just in case of save button clicked
 
-  // add event listener to maintain form updates
-  form.addEventListener('change', (event) => {
-    const { target } = event;
-    const { name } = target;
+  window.addEventListener('load',renderSafari); // add event listerner to properly render css in case browser is safari 
 
-    // get checked boolean value for checkboxes and radio buttons
-    const isCheckedType = ['checkbox', 'radio'].some(
-      (type) => type === target.type,
-    );
+  form.addEventListener('change', event => {
+    const name = event.target.name;
+    let value = event.target.value; // get checked boolean value for checkboxes
 
-    const value = isCheckedType ? target.checked : target.value;
+    if (event.target.type === 'checkbox') {
+      value = event.target.checked;
+    } //convert radio button strings to boolean if values are 'true' or 'false'
+
+
+    if (event.target.type === 'radio') {
+      if (event.target.value === 'true') {
+        value = true;
+      }
+
+      if (event.target.value === 'false') {
+        value = false;
+      }
```

Comment:
can we replace the 2 inner if statements with 
`value = event.target.value;` ?

Reply:
The value needs to be assigned to a boolean in order to properly handle the form changes.